src/electronic-ids/pcsc/EIDIDEMIA.cpp

```diff
@@ -32,13 +32,31 @@ namespace
 const byte_vector::value_type PIN_PADDING_CHAR = 0xFF;
 const byte_vector::value_type AUTH_PIN_REFERENCE = 0x01;
 
+template <typename F>
+struct ScopeExit
```

Comment:
We already have a scope guard implementation here: https://github.com/web-eid/libelectronic-id/blob/main/src/electronic-ids/pkcs11/PKCS11CardManager.hpp#L52-L54. Can we find a DRY solution that can be used in both places?